app/src/main/java/com/itsaky/androidide/activities/OnboardingActivity.kt

```diff
@@ -0,0 +1,212 @@
+
+  private val onBackPressedCallback = object : OnBackPressedCallback(true) {
+    override fun handleOnBackPressed() {
+      finishAffinity()
+    }
+  }
+
+  private val isStoragePermissionGranted: Boolean
+    get() =
+      (ContextCompat.checkSelfPermission(this, permission.READ_EXTERNAL_STORAGE) ==
+          PackageManager.PERMISSION_GRANTED &&
+          ContextCompat.checkSelfPermission(this, permission.WRITE_EXTERNAL_STORAGE) ==
+          PackageManager.PERMISSION_GRANTED)
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setTransformer(AppIntroPageTransformerType.Fade)
+    isIndicatorEnabled = true
+    isWizardMode = true
+    setProgressIndicator()
+    showStatusBar(true)
+
+    if (checkDeviceSupported()) {
+      if (!isStoragePermissionGranted) {
+        addSlide(OnboardingFragment.newInstance(
+          string.title_file_access,
+          string.msg_file_access,
+          R.raw.statistics_animation, // TODO: Replace the animation with a more appropriate one
+          name = ""
+        ))
+        askForPermissions(
+          permissions = arrayOf(
+            permission.WRITE_EXTERNAL_STORAGE,
+            permission.READ_EXTERNAL_STORAGE
+          ),
+          slideNumber = if (!archConfigWarnHasShown()) 1 else 2,
+          required = true
+        )
+      }
+      if (!statConsentDialogShown) {
+        addSlide(OnboardingFragment.newInstance(
+          string.title_androidide_statistics,
+          string.msg_androidide_statistics,
+          R.raw.statistics_animation,
+          "statistics"
+        ))
+      }
+      if (!checkToolsIsInstalled()) {
+        addSlide(OnboardingFragment.newInstance(
+          string.title_install_jdk_sdk,
+          string.msg_require_install_jdk_and_android_sdk,
+          R.raw.java_animation,
+          "install_jdk_sdk"
+        ))
+      }
+    }
+
+    onBackPressedDispatcher.addCallback(this, onBackPressedCallback)
+  }
+
+  override fun onUserDeniedPermission(permissionName: String) {
+    // User pressed "Deny" on the permission dialog
+    flashError(string.msg_storage_denied)
+  }
+
+  override fun onUserDisabledPermission(permissionName: String) {
+    // User pressed "Deny" + "Don't ask again" on the permission dialog
+  }
+
+  override fun onResume() {
+    super.onResume()
+    if (isSetupDone()) {
+      openActivity(MainActivity::class.java)
+      finish()
+    }
+  }
+
+  override fun onSlideChanged(oldFragment: Fragment?, newFragment: Fragment?) {
+    super.onSlideChanged(oldFragment, newFragment)
+    checkConsent(oldFragment)
+  }
+
+  override fun onNextPressed(currentFragment: Fragment?) {
+    super.onNextPressed(currentFragment)
+    checkConsent(currentFragment)
+  }
+
+  private fun checkConsent(fragment: Fragment?) {
+    if (fragment is OnboardingFragment) {
+      when (fragment.name) {
+        "statistics" -> {
+          statOptIn = true
+          statConsentDialogShown = true
```

Comment:
The user should have the option to opt-out of the stat collection. Add a checkbox to the fragment to allow to user to opt-out. The checkbox should be checked by default.